Plugin.php

```diff
@@ -8,28 +8,192 @@
 namespace yii\composer;
 
 use Composer\Composer;
+use Composer\DependencyResolver\Operation\UpdateOperation;
+use Composer\EventDispatcher\EventSubscriberInterface;
+use Composer\Installer\PackageEvent;
+use Composer\Installer\PackageEvents;
 use Composer\IO\IOInterface;
 use Composer\Plugin\PluginInterface;
+use Composer\Script;
+use Composer\Script\ScriptEvents;
 
 /**
  * Plugin is the composer plugin that registers the Yii composer installer.
  *
  * @author Qiang Xue <qiang.xue@gmail.com>
  * @since 2.0
  */
-class Plugin implements PluginInterface
+class Plugin implements PluginInterface, EventSubscriberInterface
 {
     /**
+     * @var array noted package updates.
+     */
+    private $_packageUpdates = [];
```

Comment:
I can not agree. There should be consistency in code style.
We claim our code style is PSR compatible.
Either we follow PSR, making exceptions **only** in case they are **necessary** or we twist PSR code standards as we like.
For example: why would not we name all private methods with `_` prefix?